### DISCUSSION TITLE
LogProbs - Incorrect?

### DISCUSSION THREAD

**Discussion by MM-IR** [vllm-project/vllm#504] [Q&A]
Hi, I am working on some multi-choice tasks in LM_evaluation_harness, where I find your results inconsistent with existing results.

- So I have some questions regarding your LogProbs specified in your SamplingParams.
- Is that correct as the log_softmax(logits)?

**Comment by zhuohan123** [comment]
It should be `log_softmax(logits)` as in the code below.

https://github.com/vllm-project/vllm/blob/bda41c70ddb124134935a90a0d51304d2ac035e8/vllm/model_executor/layers/sampler.py#L74-L79

Convert this issue to a discussion for future questions.

**Reply by MM-IR** [reply]
Oh sorry, I have noticed that. Can you support us to access the whole sentence logits/logProbs by autoregressive LMs? 

There are some multi-choice tasks, requiring us to manipulate the generated tokens, instead of letting these LMs generate the completion tokens themselves.

**Comment by RanchiZhao** [comment]
same problem. How to get ppl in vllm api?
